llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -4177,10 +4176,6 @@ class VPBlockUtils {
                             unsigned PredIdx = -1u, unsigned SuccIdx = -1u) {
     assert((From->getParent() == To->getParent()) &&
            "Can't connect two block with different parents");
-    assert((SuccIdx != -1u || From->getNumSuccessors() < 2) &&
-           "Blocks can't have more than two successors.");
-    assert((PredIdx != -1u || To->getNumPredecessors() < 2) &&
-           "Blocks can't have more than two predecessors.");
```

Comment:
These asserts no longer hold?

Reply:
The second one doesn't hold any more, e.g. scalar.ph can now have more than 2 predecessors. Restored the first one.